packages/buildcop/src/buildcop.ts

```diff
@@ -44,28 +44,33 @@ interface TestFailure {
   testCase: string;
 }
 
-interface PubSubPayload {
+export interface BuildCopPayload {
   repoOwner: string;
   repoName: string;
   buildID: string;
   buildURL: string;
-  xunitXML: string;
+  xunitXML: string; // base64 encoded to avoid JSON escaping issues.
 }
 
 interface PubSubContext {
   readonly event: string;
   github: GitHubAPI;
-  payload: PubSubPayload;
   log: LoggerWithTarget;
+  payload: {
+    payload: BuildCopPayload
+  }
 }
 
-function handler(app: Application) {
+export function buildcop(app: Application) {
   app.on('pubsub.message', async (context: PubSubContext) => {
-    const owner = context.payload.repoOwner;
-    const repo = context.payload.repoName;
-    const buildID = context.payload.buildID || '[TODO: set buildID]';
-    const buildURL = context.payload.buildURL || '[TODO: set buildURL]';
-    const xml = context.payload.xunitXML;
+    const payload: BuildCopPayload = context.payload.payload as BuildCopPayload;
+
+    const owner = payload.repoOwner;
```

Comment:
rather than setting this in your payload, you could use the approach here:

https://github.com/googleapis/repo-automation-bots/blob/master/packages/failurechecker/src/failurechecker.ts#L43

This gets populated magically by `gcf-utils`.